### ISSUE TITLE
Steam + Oculus broken in VR - Only first frame rendered

### ISSUE THREAD

**Issue by shaneharris** [webmixedreality/exokit#742]
Repro `node . -x webvr https://possible-bench.glitch.me/` aframe master from november.

Probably requires an openvr lib update - @modulesio 

Will look at this tomorrow.

**Comment by shaneharris** [comment]
Not 100 % if this is actually steam, but its the only third party here, exokit wasn't updated, aframe wasn't updated and things just stopped working. Needs more investigation

**Comment by shaneharris** [comment]
This also might be related to steam + oculus together - this is my test hardware currently. Will test vive too.

**Comment by modulesio** [comment]
I would log/windbg to see where it gets stuck, to check if it get stuck at all (it could very well not be stuck, just failing to blit).

Here's a call that's likely to prove insightful: https://github.com/webmixedreality/exokit/blob/0069e1e774b8415a4c2cc4f7bd6ff375728ba57d/src/index.js

Luckily this call path is pretty well defined/well-trodden so it shouldn't be hard to find the culprit. And the solution might be a libopenvr update.

**Comment by shaneharris** [comment]
There are no JS errors on screen, and everything loads as normal according the the JS console output, its just like you say nothing draws after the first draw call and steamvr says "(unresponsive) node.exe"

**Comment by modulesio** [comment]
Then some call (possibly that one) is likely hanging.

**Comment by shaneharris** [comment]
Windbg output doesnt show anything promising:

```
Tue Jan 15 2019 00:18:46.362 - //==============================================================================================
Tue Jan 15 2019 00:18:46.362 - ================================================================================================
Tue Jan 15 2019 00:18:46.362 - ================================================================================================
Tue Jan 15 2019 00:18:46.363 - node.exe startup with PID=13880, config=C:\Program Files (x86)\Steam\config, runtime=C:\Program Files (x86)\Steam\steamapps\common\SteamVR\
Tue Jan 15 2019 00:18:46.363 - vrclient type=VRApplication_Scene
Tue Jan 15 2019 00:18:46.369 - [Settings] Load Default Json Settings from C:\Program Files (x86)\Steam\steamapps\common\SteamVR\resources\settings\default.vrsettings
Tue Jan 15 2019 00:18:46.370 - [Settings] Load Default Json Settings from C:\Program Files (x86)\Steam\steamapps\common\SteamVR\drivers\htc\resources\settings\default.vrsettings
Tue Jan 15 2019 00:18:46.371 - [Settings] Load Default Json Settings from C:\Program Files (x86)\Steam\steamapps\common\SteamVR\drivers\lighthouse\resources\settings\default.vrsettings
Tue Jan 15 2019 00:18:46.371 - [Settings] Load Default Json Settings from C:\Program Files (x86)\Steam\steamapps\common\SteamVR\drivers\null\resources\settings\default.vrsettings
Tue Jan 15 2019 00:18:46.373 - [Settings] Load Json Settings from C:\Program Files (x86)\Steam\config\steamvr.vrsettings
Tue Jan 15 2019 00:18:49.939 - Received success response from vrserver connect
Tue Jan 15 2019 00:18:49.941 - Not looking for a good app key because Steam didn't start this app
Tue Jan 15 2019 00:18:49.941 - App key after connect message:system.generated.node.exe
Tue Jan 15 2019 00:18:49.942 - Enabling legacy input mode
Tue Jan 15 2019 00:18:49.962 - Received success response from vrcompositor connect
Tue Jan 15 2019 00:18:49.963 - Initializing the limited version of CVRCompositorClient
ModLoad: 00007ffe`1fcb0000 00007ffe`1fcd4000   C:\Windows\SYSTEM32\resourcepolicyclient.dll
Tue Jan 15 2019 00:18:50.360 - Capturing Scene Focus
ModLoad: 00007ffe`16f70000 00007ffe`17461000   C:\Windows\system32\nvapi64.dll
Tue Jan 15 2019 00:18:51.778 - Found Windows 10 or newer, so enable advanced image processing of scene textures.
Tue Jan 15 2019 00:18:51.778 - Found IDXGIAdapter index 0 which matches expected LUID 0x843b.
ModLoad: 00007ffe`1a4f0000 00007ffe`1a5de000   C:\Windows\System32\DriverStore\FileRepository\nvaki.inf_amd64_69abfa622d6a03f7\nvldumdx.dll
ModLoad: 00007ffe`24230000 00007ffe`2424d000   C:\Windows\System32\imagehlp.dll
ModLoad: 00007ffd`ea5d0000 00007ffd`ec951000   C:\Windows\System32\DriverStore\FileRepository\nvaki.inf_amd64_69abfa622d6a03f7\nvwgf2umx.dll
ModLoad: 00007ffe`027f0000 00007ffe`02875000   C:\Program Files\NVIDIA Corporation\Ansel\Tools\NvCameraWhitelisting64.dll
Tue Jan 15 2019 00:18:52.132 - Setting max texture dimensions to 2152x2562 before requiring downsampling
Tue Jan 15 2019 00:19:21.479 - PropertyContainer: High lock acquisition counters: sp:100001 yd:1 re:0
Tue Jan 15 2019 00:19:23.601 - PropertyContainer: High lock acquisition counters: sp:100001 yd:1 re:0
```

**Comment by shaneharris** [comment]
![image](https://user-images.githubusercontent.com/6830845/51149800-973da180-185b-11e9-8086-c1feaa049daf.png)

**Comment by modulesio** [comment]
It wouldn't be the windbg output, but stepping through it to find what's
hanging. If you hang it you can probably break in and use `k` to get the
hung stack trace.

**Comment by modulesio** [comment]
How is the CPU usage? Is it an actual hang, or failure to blit? I.e. does
the node loop run?

**Comment by shaneharris** [comment]
Cpu usage is nornal. Node does not hang it continues to run, yes it looks like a failure to blit

**Comment by modulesio** [comment]
I would try glGetError() to see if something is erroring.

Any GL call can be intercepted here:
https://github.com/webmixedreality/exokit/blob/0069e1e774b8415a4c2cc4f7bd6ff375728ba57d/src/core.js#L2001

And if you gl.getError() you can find out which one, and which stack,
actually errors, if it's user code. If it's not user code it could be
something in the libopenvr binding (some different framebuffer post-update
or such).

**Comment by shaneharris** [comment]
Will test this first thing tomorrow

**Comment by shaneharris** [comment]
Sorry for the delay! So these are both false in my case so it doesnt get near that block of code. - https://github.com/webmixedreality/exokit/blob/0069e1e774b8415a4c2cc4f7bd6ff375728ba57d/src/core.js#L1996

**Comment by shaneharris** [comment]
OK so i tried with `-f true` and `-m true` and i get the following error - 
```Bugsnag:  Encountered an uncaught error, terminating…
TypeError: Cannot read property 'vr' of undefined
    at HTMLElement.value (https://cdn.jsdelivr.net/gh/aframevr/aframe@231468dfc41b4cc15ffeedd28ac3f8ac8201a938/dist/aframe-master.js:75651:34)
    at bound (https://cdn.jsdelivr.net/gh/aframevr/aframe@231468dfc41b4cc15ffeedd28ac3f8ac8201a938/dist/aframe-master.js:79451:17)
    at ontimeout (timers.js:438:13)
    at tryOnTimeout (timers.js:300:5)
    at listOnTimeout (timers.js:263:5)
    at Timer.processTimers (timers.js:223:10)```


Which translates to this line - 
https://github.com/aframevr/aframe/blob/c55f024b613898fbefc03ff6c2c3ba60812c9a2c/src/core/scene/a-scene.js#L521

**Comment by chrislatorres** [comment]
Was this fixed by #754?

**Comment by modulesio** [comment]
I don't think it was. However, I would consider this issue subsumed by https://github.com/webmixedreality/exokit/pull/787. Oculus was never officially supported before that PR.
